### ISSUE TITLE
print is zero indexed in the parser

### ISSUE THREAD

**Issue by dvd101x** [josdejong/mathjs#2989]
Checking the embedded docs for the function `print` there is the following example

```print("Values: $0, $1, $2", [6, 9, 4])```

It's kind of odd since in the parser there is no zero index in matrices, maybe it should be like:

```print("Values: $1, $2, $3", [6, 9, 4])```

So my suggestion is to review the possibility for print to be one indexed in the parser and adjust it's embedded docs accordingly.

**To Reproduce**
Write in the parser
`help(print)` 

Try the following example in the parser
```print("Values: $0, $1, $2", [6, 9, 4])```

**Comment by josdejong** [comment]
Thanks for reporting, we should indeed make the function `print` one-based in the expression parser, like other functions such as `map`, `subset` and `index`. To do that, we should implement a [transform function](https://github.com/josdejong/mathjs/tree/develop/src/expression/transform) for it, which changes the indexes used in `print` from 1-based to 0-based. Anyone interested in implementing that?

**Comment by dvd101x** [comment]
Hi, while working on this and doing tests I noticed that it's expected to work with an array inside an object.

```js
math.print("I like $food.0", {food:["pizza"]})    // "I like pizza"
```

But in the parser
```jl
print("I like $food.0", {food:["pizza"]})         # "I like $food.0"
```

This seems like a separate issue, shall it be addressed in a separate PR ?

**Comment by Ayush-Kumar91221** [comment]
> Thanks for reporting, we should indeed make the function `print` one-based in the expression parser, like other functions such as `map`, `subset` and `index`. To do that, we should implement a [transform function](https://github.com/josdejong/mathjs/tree/develop/src/expression/transform) for it, which changes the indexes used in `print` from 1-based to 0-based. Anyone interested in implementing that?

i would like to try to implement it. Is this issue still open?

**Comment by Ayush-Kumar91221** [comment]
i think i have found the solution to the problem. we just need to alter the file [https://github.com/josdejong/mathjs/blob/49c793ba5a47359157a2c5009e63b11e43fcff2e/src/function/string/print.js#L1](url) and change the indices to the keys-1

**Comment by dvd101x** [comment]
I found the issue regarding arrays inside objects not working in the parser.

The issue is that `print` doesn't do the automatic conversion from matrix to array if it's part of an object. Thus this also doesn't work.

```js
math.print("I like $food.0", {food: math.matrix(["pizza"])})    // I like $food.0
```

I think it will be a quick fix and will get it ready in a PR maybe next week.

**Comment by josdejong** [comment]
Fix published now in `v11.11.1`
